commerce_coordinator/apps/stripe/views.py

```diff
@@ -66,17 +68,32 @@ def post(self, request):
             raise UnhandledStripeEventAPIError
 
         payment_intent = event.data.object
+
+        event_source_system_identifier = payment_intent.metadata.get('source_system')
         logger.info(
-            '[Stripe webhooks] event %s with amount %d and payment intent ID [%s].',
+            '[Stripe webhooks] event %s with amount %d and payment intent ID [%s], source: [%s].',
             event.type,
             payment_intent.amount,
             payment_intent.id,
+            event_source_system_identifier,
         )
 
+        if event_source_system_identifier != source_system_identifier:
+            logger.info(
+                '[Stripe webhooks] Skipping event %s with payment intent ID [%s], source: [%s].',
+                event.type,
+                payment_intent.id,
+                event_source_system_identifier,
+            )
+            return Response(status=status.HTTP_200_OK)
```

Comment:
Here I have added a check to make sure we only process events that were created by Commerce Coordinator only.